packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -358,7 +360,7 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
         className={ 'ms-BaseButton-menuhost ' + menuProps.className }
         target={ this._isSplitButton ? this._splitButtonContainer : this._buttonElement }
         labelElementId={ this._labelId }
-        onDismiss={ this._onToggleMenu }
+        onDismiss={ onDismiss }
```

Comment:
Hmm. You would now be skipping the logic in onToggleMenu. Are you sure that is desired?

Reply:
Sorry, I was going to add a little bit to this. PR incoming.

Reply:
This will still not address the issue to not dismiss onClick -

![image](https://user-images.githubusercontent.com/32884685/32020561-7248cef8-b985-11e7-8f95-e5d8533cb28c.png)

Reply:
@avnikp correct, which is why I made #3225

Reply:
ah, thanks! I'll take a look.